internal/extsvc/github/common.go

```
@@ -1552,13 +1552,22 @@ func doRequest(ctx context.Context, apiURL *url.URL, auth auth.Authenticator, ra
 		rateLimitMonitor.Update(resp.Header)
 	}
 
-	if resp.StatusCode < 200 || resp.StatusCode >= 400 {
+	if resp.StatusCode < 200 || (resp.StatusCode >= 400) {
 		var err APIError
+
 		if body, readErr := io.ReadAll(io.LimitReader(resp.Body, 1<<13)); readErr != nil { // 8kb
 			err.Message = fmt.Sprintf("failed to read error response from GitHub API: %v: %q", readErr, string(body))
 		} else if decErr := json.Unmarshal(body, &err); decErr != nil {
 			err.Message = fmt.Sprintf("failed to decode error response from GitHub API: %v: %q", decErr, string(body))
 		}
+
+		if resp.StatusCode >= 500 && outageChecker != nil {
+			down, _ := outageChecker.IsServiceDown(ctx) // TODO: is it safe to ignore this error?
+			if down {
```

Would  it be safe to ignore this error? 🤔

---

It is probably always a good idea to check the error and at least log it. In this case the behaviour of the code probably doesn't change much depending on the result, but I would still log it in case it contains extra information when it comes to debugging.